<launch>
  <arg name="teleop_joy" default="false" />
  <node pkg="bw_io" type="bw_io_node" name="bw_io_serial_driver">
    <param name="port" value="/dev/bw_io"/>
    <param name="baud" value="115200"/>
    <!-- kx 代表模拟通道的斜率 bx代表模拟通道的截距 mvalue_x 代表通道最小值过滤阈值 {-mvaluex mvalueX}范围内的值会被设成零 -->
    <param name="k1" value="1.0"/>
    <param name="b1" value="0.0"/>
    <param name="mvalue1" value="0.1"/>
    <param name="k2" value="1.0"/>
    <param name="b2" value="0.0"/>
    <param name="mvalue2" value="0.1"/>
    <param name="k3" value="1.0"/>
    <param name="b3" value="0.0"/>
    <param name="mvalue3" value="0.1"/>
    <param name="k4" value="1.0"/>
    <param name="b4" value="0.0"/>
    <param name="mvalue4" value="0.1"/>
  </node>
  <node if="$(arg teleop_joy)" pkg="bw_io" type="xiaoqiang_teleop_joy" name="xiaoqiang_teleop_joystick">
    <param name="scale_angular" value="0.3"/>
    <param name="scale_linear" value="0.5"/>
    <param name="axis_linear" value="2"/>
    <param name="axis_angular" value="1"/>
    <param name="axis_deadman" value="1"/>
    <param name="axis_up" value="2"/>
    <param name="axis_down" value="3"/>
    <param name="enable_updown" value="false"/>
  </node>
</launch>
